### PR TITLE
Slime console

### PR DESCRIPTION
**Pull request by Groudonmaster**
### Adds Slime management console

Shameless TG port

A free look/advanced camera console that can pick up and place slimes remotely, as well as deploy monkeys from stored monkey cubes. It can also pick up dead monkeys to recycle them directly into the machine. One of the biggest complaints about xenobio is how tedious and clunky it is dragging all the monkeys and slimes about, so hopefully this makes it more enjoyable.

The eye can only move inside the Xenobiology Lab so you can't use it to hunt badguys or dump slimes everywhere. However renaming areas to Xenobiology Lab will allow you to access them if you want to steal the blueprints and expand your territory.

DONE:
-Basically the whole machine. All the functions of it work in testing.
-Restricted to xenobiology only
-Icons
-Cerulean slimes now have a new reaction which gives you one use blueprints, for allowing you to expand your territory.

Thanks to @AnturK for making these advanced camera machines in the first place

Ports @KorPhaeron Slime management console code.

NEEDS TO BE ADDED TO THE MAP.

### CODE REVIEW

**Comment by Groudonmaster** [comment]
Also this was all done via web so I do need to test it still. Or if someone else wants to they can.

**Comment by Groudonmaster** [comment]
What's failing here. I'm retarded.

**Comment by Cruix** [comment]
You didn't update the .dme, so it isn't looking at code/modules/research/xenobiology/xenobio_camera.dm when it compiles. Open your branch In DM and checkmark the file in the hierarchy and it should work. Or if it compiles on your machine, just commit the changes to the .dme.

**Comment by oisin100** [comment]
> You didn't update the .dme, so it isn't looking at code/modules/research/xenobiology/xenobio_camera.dm when it compiles. Open your branch In DM and checkmark the file in the hierarchy and it should work.

Mhm, But Groudon is a scrub and uses web github.

File: code/modules/research/xenobiology/xenobio_camera.dm [hunk 1]
```diff
@@ -0,0 +1,163 @@
+//Xenobio control console
+/mob/camera/aiEye/remote/xenobio
+	visible_icon = 1
+	icon = 'icons/obj/abductor.dmi'
+	icon_state = "camera_target"
+
+
+/mob/camera/aiEye/remote/xenobio/setLoc(var/t)
+	var/area/new_area = get_area(t)
+	if(new_area.name && new_area.name == "Xenobiology Lab")
```
Comment (Cruix) [inline code/modules/research/xenobiology/xenobio_camera.dm:10]: Make this line
if(new_area && initial(new_area.name) == "Xenobiology Lab")
so you can't use the CE's blueprints to teleport slimes all over the station.

File: html/changelogs/KOR-SLIME.yml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+#   rscdel (general deleting of nice things)
+#   imageadd
+#   imagedel
+#   spellcheck (typo fixes)
+#   experiment
+#   tgs (TG-ported fixes?)
+#################################
+
+# Your name.  
+author: Kor
```
Comment (imtakingabreakdontatme) [inline html/changelogs/KOR-SLIME.yml:25]: Huh, I think this is the first time anyone has actually credited in the changelog after porting it instead of putting their name on it. Thanks. Let me know if you need any help with this.

File: code/modules/research/xenobiology/xenobio_camera.dm (hunk 1 above)
Comment (Groudonmaster) [inline code/modules/research/xenobiology/xenobio_camera.dm:10]: I think that was the intention. Just another way traitors can be creative.

EDIT: It says in the PR description that was intended :p

**Comment by Groudonmaster** [comment]
@KorPhaeron I'm having an issue where if you interact with the console it will just do this (refer to picture) and you won't be able to disengage with it.

xenobio

none of the actions work.

EDIT: Character is not actually moveable. But then you're just trapped at the console.

**Comment by imtakingabreakdontatme** [comment]
Uh, where'd the cancel button go? There should be a cancel action at the top

**Comment by Groudonmaster** [comment]
Your guess is as good as mine. It turned it into "Generic action"

**Comment by imtakingabreakdontatme** [comment]
Do you guys have abductors?

**Comment by Groudonmaster** [comment]
Yeah but that code may have changed. I know a bugfix was done recently with it involving the console not working properly

**Comment by imtakingabreakdontatme** [comment]
I'd try making sure you are up to date with abductor cameras first, yeah

**Comment by X-TheDark** [comment]
This is (probably) because of us not having /datum/action/innate/ type buttons (we are still using the old action_type system for that), which was implemented in https://github.com/tgstation/-tg-station/pull/12597

**Comment by oisin100** [comment]
This will be done in the rebase.
